### PR TITLE
Update How shamirs add and remove lore

### PR DESCRIPTION
**Pull request by hozz8528**
Made it so shamirs prepend their lore to existing lore when added, updated removing shamirs so that it copies the lore to storage, finds the "Shamir" line and removes that and the surrounding lines before applying it back to the item, and also updated the way the finalized shamir item gets the lore from the item so that it gets the type of shamir in a similar way to how I remove specific lore.

Basically, adding and removing shamirs won't mess up your lore now.

### CODE REVIEW

**Comment by BluePsychoRanger** [comment]
#576 does this in a less specific way, utilizing a new library dealing with lore

**Comment by hozz8528** [comment]
closing since lib_lore makes this unnecessary
